x-pack/plugins/maps/public/connected_components/mb_map/tooltip_control/tooltip_control.tsx

```diff
@@ -178,7 +178,7 @@ export class TooltipControl extends Component<Props, {}> {
     for (let i = 0; i < mbFeatures.length; i++) {
       const mbFeature = mbFeatures[i];
       const layer = this._getLayerByMbLayerId(mbFeature.layer.id);
-      if (!layer) {
+      if (!layer || !layer.isShowTooltip()) {
```

Comment:
IVectorLayer has an existing method `canShowTooltip`. Used below.  How about updating IVectorLayer.canShowTooltip to include check for showTooltip and use canShowTooltip instead of isShowTooltip here and then remove canShowTooltip usage below. This is a better place as it short circuits the logic sooner.

Reply:
`canShowTooltip` should not rely on `areTooltipsDisabled` because `canShowTooltip` determines whether the toggle switch element is [disabled](https://github.com/elastic/kibana/pull/138275/files#diff-4dbb09066d854a32e834bec442178363e8da72c20bfb53db251600e76d727bc7R182) in the Layer settings.